src/Akka.Streams.Kafka.Tests/Integration/RebalanceIntegrationTests.cs

```diff
@@ -22,110 +28,211 @@ public RebalanceIntegrationTests(ITestOutputHelper output, KafkaFixture fixture)
     {
     }
 
-    private static Source<CommittableMessage<Null, string>, IControl> GetConsumer(
-        ConsumerSettings<Null, string> settings,
-        string topic)
-    {
-        return KafkaConsumer.CommittableSource(settings, Subscriptions.Topics(topic));
-    }
-
-    private sealed class StreamCompleted
-    {
-        public static StreamCompleted Instance { get; } = new();
-
-        private StreamCompleted()
-        {
-        }
-    }
-
-    private sealed record StreamFailed(Exception Ex);
-
-    private IKillSwitch CreateKillableStream(string topic, ConsumerSettings<Null, string> settings, IActorRef sinkRef, string consumerName)
-    {
-        var source = GetConsumer(settings, topic);
-        var killSwitch = source
-            .WithAttributes(Attributes.CreateName(consumerName))
-            .ViaMaterialized(KillSwitches.Single<CommittableMessage<Null, string>>(), Keep.Right)
-            .SelectAsync(1, async msg =>
-            {
-                // commit the offset
-                await msg.CommitableOffset.Commit();
-                return msg.Record;
-            })
-            .WithAttributes(Attributes.CreateName($"selectAsync-{consumerName}"))
-            .ToMaterialized(
-                Sink.ActorRef<ConsumeResult<Null, string>>(sinkRef, StreamCompleted.Instance,
-                    exception => new StreamFailed(exception)), Keep.Left)
-            .Run(Materializer.WithNamePrefix(consumerName));
-
-        return killSwitch;
-    }
+    private static readonly IReadOnlyList<int> Numbers = Enumerable.Range(0, 5000).ToList();
+    private const string ConsumerClientId1 = "consumer-1";
+    private const string ConsumerClientId2 = "consumer-2";
 
     /// <summary>
     /// Reproduction spec for https://github.com/akkadotnet/Akka.Streams.Kafka/issues/415
     /// </summary>
     [Fact]
-    public async Task ShouldReBalanceWithoutArgumentExceptions()
+    public async Task FetchedRecords_must_be_removed_from_source_stage_buffer_when_partition_is_removed()
```

Comment:
Spec has been rewritten to more or less match the rebalance spec - and in this case it's checking to ensure that the individual stream stages do their level best to filter out messages from revoked partitions before emitting them downstream. This is all with the intention of eliminating duplicates, as those messages will be picked up by someone else.